walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,121 @@
 
 package com.google.sps;
 
+import java.util.Arrays;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+import java.lang.*;
 
 public final class FindMeetingQuery {
+  /** MeetingRequest request has:
+      getAttendees() method which returns hashSet of mandatory attendees
+      getDuration() method which tells us how long our meeting has to be */
+
+  /** Collection<Event> events has events. Each event has:
+      getWhen() method which tells us time range of event
+      getAttendees() method which tells us who has to go to each event */
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> mandatoryAttendees = request.getAttendees();
+
+    // Check edge cases
+    MeetingRequest requestAllMandatory = new MeetingRequest (mandatoryAttendees, request.getDuration());
+    ArrayList<TimeRange> solutions = new ArrayList<>();
+
+    // Requests with duration longer than a day, return no solutions. 
+    if (requestAllMandatory.getDuration() > TimeRange.WHOLE_DAY.duration()){
+      return solutions;
+    }
+
+    // If there are no attendees or meeting duration is 0, return whole day as solution. 
+    if (requestAllMandatory.getAttendees().isEmpty() || requestAllMandatory.getDuration() == 0){
+      solutions.add(TimeRange.WHOLE_DAY);
+      return solutions;
+    }
+
+    ArrayList<TimeRange> relevantTimes = new ArrayList<>();
+
+    // Iterate through all events, ensuring each event has at least one member from mandatoryAttendees    
+    Iterator<Event> iter = events.iterator();
+    while (iter.hasNext()) {
+      Event event = iter.next();
+      Set<String> attendees = event.getAttendees();
+      boolean overlap = false;
+
+      for (String attendee : attendees) {
+        if (mandatoryAttendees.contains(attendee)){
+          overlap = true; // overlap is true if this event has a mandatory attendee
+          break;
+        }
+      }
+
+      if (overlap) {
+        relevantTimes.add(event.getWhen());
```

Comment:
You could actually just add this line above the "break" statement and get rid of the "overlap" variable